swupd-inspector/main.go

```diff
@@ -0,0 +1,208 @@
+package main
+
+import (
+	"bufio"
+	"fmt"
+	"log"
+	"os"
+	"os/user"
+	"path/filepath"
+	"strconv"
+	"strings"
+)
+
+// TODO: Flag to set cacheDir.
+
+// TODO: Use XDG_* environment variables instead of ~/.config and ~/.cache by default.
+
+// TODO: Take into account deleted files, right now 'get' fails trying to download 0000...0.tar.
+
+func usage() {
+	fmt.Printf(`swupd-inspector analyzes swupd content
+
+Commands:
+
+    cat URL Manifest.NAME
+        Print the contents of the given Manifest.
+
+    get URL Manifest.NAME
+        Download the contents of the given Manifest.
+
+    get URL FILENAME
+        Search for the FILENAME and download the corresponding fullfile.
+        The FILENAME must be an absolute path.
+
+    get URL HASH
+        Download the fullfile corresponding to the hash.
+
+    diff [--strict] [--no-color] URL1 URL2
+        Compare two versions of swupd content. The filenames and flags
+        will be compared, recursing to the bundles. Use --strict to
+        also compare the version numbers of the files. Use --no-color
+        to not emit escape codes in the output.
+
+    log URL FILENAME
+        Show FILENAME version and its previous versions.
+
+    clean
+        Clean up any cached content.
+
+The program will cache everything downloaded, and can keep content from
+different sources. The cache directory is $HOME/.cache/swupd-inspector.
+
+The URLs must refer to a specific version like
+https://download.clearlinux.org/update/20520. Absolute local paths can
+also be used instead of URLs.
+
+It is possible to refer to content by aliases. The alias 'clear' works
+by default, so clear/20520 refer to the same as the URL above. Other
+aliases can be defined in $HOME/.config/swupd-inspector/aliases in the
+format ALIAS=URL per line.
+`)
+}
+
+func main() {
+	log.SetFlags(0)
+
+	if len(os.Args) < 2 {
```

Comment:
Ok, I would *much* prefer you use `cobra` here to manage your CLI. This is not a simple CLI and using cobra will help organize it a lot better. It will also keep your usage text consistent with the other main `mixer-tools` binaries, `mixer` and `mixin`.

Reply:
Will do this.

Reply:
Done.